packages/vc-export/src/KiltAttestationProofV1.ts

```diff
@@ -0,0 +1,668 @@
+  const apiAt = await api.at(blockHash)
+  // TODO: should we look up attestation storage as well and take attestation info from there? That gives us the definitive attestation state in this block (e.g. makes sure it hasn't been removed after)
+  const [events, time] = await apiAt.queryMulti<
+    [Vec<FrameSystemEventRecord>, u64]
+  >([
+    [api.query.system.events as QueryableStorageEntry<'promise'>],
+    [api.query.timestamp.now as QueryableStorageEntry<'promise'>],
+  ])
+
+  const timestamp = time.toNumber()
+  const attestationEvent = events
+    .reverse()
+    .find(
+      ({ phase, event }) =>
+        phase.isApplyExtrinsic &&
+        api.events.attestation.AttestationCreated.is(event) &&
+        u8aEq(event.data[1], claimHash)
+    )
+  if (!attestationEvent)
+    throw new SDKErrors.CredentialUnverifiableError(
+      `Matching attestation event for root hash ${u8aToHex(
+        claimHash
+      )} not found at block ${u8aToHex(blockHash)}`
+    )
+  const [att, , cTypeHash, authorization] = attestationEvent.event.data as [
+    AccountId,
+    Hash,
+    Hash,
+    Option<Hash> | Option<RuntimeCommonAuthorizationAuthorizationId>
+  ] &
+    IEventData
+  const attester = getFullDidUri(encodeAddress(att.toU8a(), 38))
+  const cTypeId = CType.hashToId(cTypeHash.toHex())
+  const delegationId = authorization.isSome
+    ? (
+        (authorization.unwrap() as RuntimeCommonAuthorizationAuthorizationId)
+          .value ?? authorization.unwrap()
+      ).toHex()
+    : null
+  return {
+    verified: true,
+    timestamp,
+    attester,
+    cTypeId,
+    delegationId,
+  }
+}
+
+async function findDelegators(
+  nodeId: string,
+  delegators: Set<DidUri>
+): Promise<void> {
+  const node = await DelegationNode.fetch(nodeId)
+  delegators.delete(node.account)
+  if (delegators.size === 0) {
+    return
+  }
+  if (node.parentId) {
+    await findDelegators(node.parentId, delegators)
+    return
+  }
+  throw new SDKErrors.CredentialUnverifiableError(
+    `The following delegators are not in the attestation's delegation hierarchy: ${[
+      ...delegators,
+    ]}`
+  )
+}
+
+/**
+ * Verifies a KILT attestation proof by querying data from the KILT blockchain.
+ * This includes querying the KILT blockchain with the credential id, which returns an attestation record if attested.
+ * This record is then compared against attester address and delegation id (the latter of which is taken directly from the credential).
+ *
+ * @param credentialInput Verifiable Credential to verify proof against.
+ * @param proof KiltAttestationProofV1 proof object to be verified. Any proofs embedded in the credentialInput are stripped and ignored.
+ * @param opts Additional parameters.
+ * @param opts.api A polkadot-js/api instance connected to the blockchain network on which the credential is anchored.
+ * @returns Object indicating whether proof could be verified.
+ */
+export async function verifyProof(
```

Comment:
Should this function also be renamed to `verify` only?

Reply:
https://github.com/KILTprotocol/sdk-js/commit/f2f6147b0c068b48e3b936f1f5b99764d5ab53f9